### DISCUSSION TITLE
Where can I find the accurate document for sootup?

### DISCUSSION THREAD

**Discussion by 1phan** [soot-oss/SootUp#688] [Q&A] [answered]
I followed the sample code in the official document and I met errors on every step.

Here is the website: https://soot-oss.github.io/SootUp/getting-started/#creating-a-view

For example, the code in doc:

```
import sootup.core.Project;
Project project = JavaProject.builder(language).addInputLocation(inputLocation).build();
JavaView view = project.createView();
```

Error:

```
error: cannot find symbol
        JavaView view = project.createView();
                               ^
  symbol:   method createView()
  location: variable project of type Project
```

The code in doc:

```
AnalysisInputLocation<JavaSootClass> inputLocation =
                PathBasedAnalysisInputLocation.createForClassContainer(pathToBinary);
```

The error:

```
error: cannot find symbol
                PathBasedAnalysisInputLocation.createForClassContainer(pathToBinary);
                                              ^
  symbol:   method createForClassContainer(Path)
  location: class PathBasedAnalysisInputLocation
```

**Comment by JonasKlauke** [comment]
create view was introduced in version 1.1 you are probably using version 1.0

createForClassContainer(pathToBinar) was not updated to the API changes.
PathBasedAnalysisInputLocation.create(pathToBinary, SourceType.Application) has replaced the old method.

I will update the doc

**Comment by swissiety** [comment, marked as answer]
please use `new JavaClassPathAnalysisInputLocation(...)` to access the underlying `PathBasedAnalysisInputLocation.create(...)` functionality.

**Reply by JonasKlauke** [reply]
okay I was not sure if there was a different 😄 I usally dont use PathBased... but i thought maybe this should be used if there is no jar. But yeah JavaClassPath.. is the way to go
